src/language-js/utils.js

```diff
@@ -12,39 +12,11 @@ const {
 const handleComments = require("./comments");
 
 /**
- * @typedef {import("@babel/types").SequenceExpression} SequenceExpression
- *
- * @typedef {import("@typescript-eslint/types").TSESTree.Literal} Literal
- *
- * @typedef {import("@babel/types").RegExpLiteral} RegExpLiteral
- * @typedef {import("@babel/types").BigIntLiteral} BigIntLiteral
- *
- * @typedef {import("@babel/types").DecimalLiteral} DecimalLiteral
- * @typedef {import("@babel/types").BooleanLiteral} BooleanLiteral
- * @typedef {import("@babel/types").NullLiteral} NullLiteral
- * @typedef {import("@babel/types").NumericLiteral} NumericLiteral
- * @typedef {import("@babel/types").StringLiteral} StringLiteral
- * @typedef {import("@babel/types").Identifier} Identifier
- * @typedef {import("@babel/types").ThisExpression} ThisExpression
- * @typedef {import("@babel/types").Super} Super
- * @typedef {import("@babel/types").PrivateName} PrivateName
- * @typedef {import("@babel/types").ArgumentPlaceholder} ArgumentPlaceholder
- * @typedef {import("@babel/types").Import} Import
- * @typedef {import("@babel/types").TemplateLiteral} TemplateLiteral
- * @typedef {import("@babel/types").ObjectExpression} ObjectExpression
- * @typedef {import("@babel/types").ArrayExpression} ArrayExpression
- *
- * @typedef {import("estree").ImportExpression} ImportExpression
- *
- * @typedef {import("@babel/types").CallExpression} CallExpression
- * @typedef {import("@babel/types").OptionalCallExpression} OptionalCallExpression
- * @typedef {import("@babel/types").NewExpression} NewExpression
- * @typedef {import("@babel/types").MemberExpression} MemberExpression
- * @typedef {import("@babel/types").OptionalMemberExpression} OptionalMemberExpression
- * @typedef {import("@babel/types").UnaryExpression} UnaryExpression
- * @typedef {import("@babel/types").TSNonNullExpression} TSNonNullExpression
- *
- * @typedef {Literal | BigIntLiteral | RegExpLiteral | DecimalLiteral | BooleanLiteral | NullLiteral | NumericLiteral | StringLiteral | Identifier | ThisExpression | Super | PrivateName | ArgumentPlaceholder | Import | TemplateLiteral | ObjectExpression | ArrayExpression | ImportExpression | CallExpression | OptionalCallExpression | NewExpression | MemberExpression | OptionalMemberExpression | UnaryExpression | TSNonNullExpression} ESNode
+ * @typedef {import("estree").Node} ESTreeNode
+ * @typedef {import("@babel/types").Node} BabelNode
+ * @typedef {import("@typescript-eslint/types").TSESTree.Node} TSNode
+ * @typedef {import("angular-estree-parser/lib/types").NGNode} NGNode
+ * @typedef {ESTreeNode | BabelNode | TSNode | NGNode} Node
```

Comment:
I wounder if we can union every ESNode in `estree.d.ts` and do `import('./estree.d.ts').Node`.

Something like

```js
import * as ESTree from 'estree'
import * as Babel from '@babel/types'
...

export type Node = ESTree.Node | Babel.Node | ...;

// A automatic loop would be nice.

```

Reply:
TBH I am not so sure how much I like the idea of mixing union of Node types from all different estree-like type packages. It feels a little sloppy to me.

I think the obvious root is that *internally* we have some kind of Frankenstein-like hybrid of ES tree node types that come in from the various parsers and go through our language-js processing.

I would favor getting a solution as clean and simple as we can for now and look into how we can improve the formal interface at some point in the future.

I am now working to see if I can resolve the `@ts-ignore` item.

Reply:
With the following change we can resolve & get rid of the `@ts-ignore`:

```diff
- * @typedef {ESTreeNode | BabelNode | TSNode | NGNode} Node
+ * @typedef {BabelNode | TSNode | NGNode} Node
```

I will do this and a little more cleanup now.

Reply:
> With the following change we can resolve & get rid of the `@ts-ignore`:
> 
> ```diff
> - * @typedef {ESTreeNode | BabelNode | TSNode | NGNode} Node
> + * @typedef {BabelNode | TSNode | NGNode} Node
> ```
> 
> I will do this and a little more cleanup now.

I don't think it's right, `estree` Literal can have `regex` property